client/src/Components/pages/UserDetails.js

```diff
@@ -1,52 +1,235 @@
-import React, { useContext } from "react";
-import { User } from "../../App";
+import { useScrollTrigger } from "@material-ui/core";
+import React, { useState, useContext } from "react";
+import { useHistory } from "react-router-dom";
+import { Authentication } from "../../App";
 
-function UserDetails() {
-  const { user } = useContext(User);
-  console.log("user:", user);
+function UserDetails({ user, setUser, setCart }) {
+  const { setIsAuth } = useContext(Authentication);
+  const history = useHistory();
+  const [editable, setEditable] = useState(false);
 
+  const updateUser = async () => {
+    const requestOptions = {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify({
+        firstName: user.name.firstName,
+        lastName: user.name.lastName,
+        phone: user.phone,
+        email: user.email,
+      }),
+    };
+    const result = await (
+      await fetch("/editUserDetails", requestOptions)
+    ).json();
+    if (result.response === false) {
+      alert("Could not update!");
+      setIsAuth(false);
+      setCart([]);
+      history.push("/login");
+    }
+  };
+
+  const cancelOrder = async (orderId) => {
+    const requestOptions = {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify({
+        orderId: orderId,
+      }),
+    };
+    const result = await (await fetch("/cancelOrder", requestOptions)).json();
+    console.log("response:", result.response);
+    if (result.response) {
+      let ordersList = user.orders;
+      ordersList = ordersList.filter((orderElement) => {
+        if (orderElement._id !== orderId) return orderElement;
+      });
+      setUser({
+        ...user,
+        orders: ordersList,
+      });
+    } else {
+      alert("Could not cancel order!");
+      history.push("/login");
+    }
+  };
   return (
-    <div className="jumbotron container">
-      <h1>User Details</h1>
-      <br />
-      <form className="userform">
-        <div className="form-group">
-          <h5>Personal Information</h5>
-          <h6 style={{ color: "blue" }} onClick={() => {}}>
-            Edit
-          </h6>
-          <input
-            type="text"
-            className="mr-2"
-            value={user.name.firstName}
-            name="FirstName"
-            disabled
-          />
-          <input
-            type="text"
-            value={user.name.lastName}
-            name="LastName"
-            disabled
-          />
-        </div>
-        <br />
-        <div className="form-group">
-          <h5>Email address</h5>
-          <h6 style={{ color: "blue" }} onClick={() => {}}>
-            Edit
-          </h6>
-          <input type="text" value={user.email} name="Email" disabled />
+    <>
+      <div className="product flex-container userbackground">
```

Comment:
userbackground can be like user-background. Try follow snake case for CSS classes